### ISSUE TITLE
My operating OS is Ubuntu 24.04, how can I use this linux driver in Ubuntu?

### ISSUE THREAD

**Issue by liaoweiyang2017** [ty2/goodix-gt7868q-linux-driver#5]

**Comment by ty2** [comment]
You can easily install it using DKMS.

```
# Install the DKMS configuration file to the appropriate directory
sudo install -Dm644 dkms/dkms.conf /usr/src/goodix-gt7868q-0.1.0/dkms.conf

# Copy the source to the DKMS source directory
sudo cp -r . /usr/src/goodix-gt7868q-0.1.0

# Add the DKMS module to the system
sudo dkms add -m goodix-gt7868q/0.1.0

# Install the DKMS module
sudo dkms install goodix-gt7868q/0.1.0

# Install the custom libinput quirks file to the appropriate directory
sudo install -Dm644 local-overrides.quirks /usr/share/libinput/60-custom-thinkbookg6p2024imh.quirks

# Check the status of the DKMS modules
sudo dkms status
```

```
# Reboot the system to apply changes
sudo reboot
```

**Comment by greenisms** [comment]
I am using a ThinkBook 16p laptop with Ubuntu 22.04 installed. I have followed the provided instructions, and everything appears to have been executed correctly. The status of the DKMS modules is displayed as 'goodix-gt7868q/0.1.1, 5.15.0-126-generic, x86_64: installed'. However, my touchpad remains unresponsive. Could you please assist me in resolving this issue?

**Comment by spyhawk** [comment]
Thank you very much for this decision! one question: how to make a right-click with a touchpad?

**Comment by ty2** [comment]
> I am using a ThinkBook 16p laptop with Ubuntu 22.04 installed. I have followed the provided instructions, and everything appears to have been executed correctly. The status of the DKMS modules is displayed as 'goodix-gt7868q/0.1.1, 5.15.0-126-generic, x86_64: installed'. However, my touchpad remains unresponsive. Could you please assist me in resolving this issue?

Did you get the quirks installed and restart the laptop?

`sudo make install-quirks`

**Comment by CiandtDyanz** [comment]
> > I am using a ThinkBook 16p laptop with Ubuntu 22.04 installed. I have followed the provided instructions, and everything appears to have been executed correctly. The status of the DKMS modules is displayed as 'goodix-gt7868q/0.1.1, 5.15.0-126-generic, x86_64: installed'. However, my touchpad remains unresponsive. Could you please assist me in resolving this issue?
> 
> Did you get the quirks installed and restart the laptop?
> 
> `sudo make install-quirks`

yes I get the quirks installed. But still can not move through my touch pad

**Comment by thanhhuynhduy** [comment]
> > I am using a ThinkBook 16p laptop with Ubuntu 22.04 installed. I have followed the provided instructions, and everything appears to have been executed correctly. The status of the DKMS modules is displayed as 'goodix-gt7868q/0.1.1, 5.15.0-126-generic, x86_64: installed'. However, my touchpad remains unresponsive. Could you please assist me in resolving this issue?
> 
> Did you get the quirks installed and restart the laptop?
> 
> `sudo make install-quirks`

I cant run this command. It's show:
make: *** No rule to make target 'install-quirks'.  Stop.

**Comment by clu3** [comment]
I'm also using ubuntu 24 LTS, I tried to install the latest kernel 6.17 hoping the patch would be included but it's not. And now when i tried to compile, it's not working, some functions like del_timer() are deprecated in the latest gcc 14
